Remove mgmt cluster check during bridge vlan add/del

rrajendran17 opened the pull request:
**Problem:**
During Harvester installation, the system attempts to add all VLAN IDs to the management bond/bridge. With Mellanox ConnectX-6 NICs (which support 512 hardware-offloaded VLANs and have hardware VLAN offloading enabled by default), this causes boot log warnings and prevents management VLAN 2301 from being added to the hardware offloaded VLAN list.

**Solution:**
dynamically add vlans to mgmt cluster network whenever user creates and updates the nad with mgmt cluster network
This will add vlan associated with that nad to mgmt-bo bridge in linux.

**Related Issue:**
https://github.com/harvester/harvester/issues/7650

**Test plan:**
1.Create NAD with vlan-id and attach it to cluster network mgmt
2.verify if bridge vlan show lists the vlan in the mgmt-bo
3.verify traffic between VMs connected over the vlan on mgmt-bo and its successful
4.Delete NAD with vlan-id from mgmt cluster
5.Verify if bridge vlan show removed vlan-id from mgmt-bo

This PR should be merged along with https://github.com/harvester/harvester-installer/pull/1005

rrajendran17 commented:
Note: since custom vlan added to mgmt-bo during harvester installation should not be removed at any point of time, I am planning to add code changes during Register() of nad controller and fetch and store the custom vlan configured by user during installation.
And a check could be added during RemoveLocalArea() for that vid and it will be removed from mgmt-bo.

I am yet to add those changes to this PR.Please let me know if that looks fine or any suggestions for better way to do this ?

w13915984028 reviewed (dismissed):
LGTM, thanks.

Meanwhile, do you have a place to record which VID is from mgmt installation? When adding other vm network/storagenetwork on mgmt network, a check is needed to avoid conflict with this VID.

At the moment, the mgmt clusternetwork has no vlanconfig, and the installation configuration is not updated to clusternetwork/vlanconfig. New enhancements are needed to unify the mgmt clusternetwork same as others.  E.g. a node controller converts mgmt network related config into a vlanconfig.

There are some comments and related specific processing on mgmt network on PR https://github.com/harvester/network-controller-harvester/pull/149.

rrajendran17 commented:
> LGTM, thanks.
> 
> Meanwhile, do you have a place to record which VID is from mgmt installation? When adding other vm network/storagenetwork on mgmt network, a check is needed to avoid conflict with this VID.
> 
> At the moment, the mgmt clusternetwork has no vlanconfig, and the installation configuration is not updated to clusternetwork/vlanconfig. New enhancements are needed to unify the mgmt clusternetwork same as others. E.g. a node controller converts mgmt network related config into a vlanconfig.
> 
> There are some comments and related specific processing on mgmt network on PR #149.

Yes I am planning this as mentioned in https://github.com/harvester/network-controller-harvester/pull/156#issuecomment-2702467664. Please let me know if there are some better options within network-controller for now until we go with above enhancement.

rrajendran17 commented:
I have added changes for storing the mgmt vlan configured during installation and added checks so that we do not remove this vlan from mgmt-bo bridge accidentally.  

@w13915984028 @starbops @mingshuoqiu Please review whenever you get time, Thanks

w13915984028 commented on pkg/network/iface/link.go:
```diff
@@ -207,3 +208,23 @@ func (l *Link) Remove() error {
+	}
+
+	for _, link := range links {
+		if !strings.Contains(link.Attrs().Name, "mgmt-br.") {
```
there are two consts to construct `mgmt-br`:
ManagementClusterNetworkName
BridgeSuffix

rrajendran17 replied:
sure, will change them

w13915984028 commented on pkg/network/iface/link.go:
```diff
@@ -207,3 +208,23 @@ func (l *Link) Remove() error {
+		}
+	}
+
+	return vlanID, nil
```
if mgmt does not configure a VID, will this function:

return 0, nil ?

rrajendran17 replied:
yes, do you think we need to configure mgmt vlan with vid 1 in that case ?

w13915984028 reviewed (commented):
need to check the upgrade path:

when PODs are upgraded, but NODEs are still not,  will the new controller work on legacy node ?
